src/resources/2022/DesignerSweatpants.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  availableAmount,
+  Skill,
+  useSkill as useSkill_,
+  visitUrl,
+} from "kolmafia";
+import { get } from "../../property.js";
+import { $item, $skill } from "../../template-string.js";
+
+/** designer sweatpants */
+export const item = $item`designer sweatpants`;
+
+/** designer sweatpants sweat skills */
+export const skills = {
+  SweatFlick: $skill`Sweat Flick`,
+  SweatSpray: $skill`Sweat Spray`,
+  SweatFlood: $skill`Sweat Flood`,
+  SipSomeSweat: $skill`Sip Some Sweat`,
+  SweatSip: $skill`Sweat Sip`,
+  DrenchYourselfInSweat: $skill`Drench Yourself in Sweat`,
+  SweatOutSomeBooze: $skill`Sweat Out Some Booze`,
+  MakeSweatAde: $skill`Make Sweat-Ade`,
+} as const;
+
+/** Map of skill to sweat cost */
+const sweatSkillCost = new Map<Skill, number>([
+  [skills.SweatFlick, 1],
+  [skills.SweatSpray, 3],
+  [skills.SweatFlood, 5],
+  [skills.SipSomeSweat, 5],
+  [skills.SweatSip, 5],
+  [skills.DrenchYourselfInSweat, 15],
+  [skills.SweatOutSomeBooze, 25],
+  [skills.MakeSweatAde, 50],
+]);
+
+/**
+ * Do you own designer sweatpants
+ * @returns Whether you have the designer sweatpants
+ */
+export function have(): boolean {
+  return availableAmount(item) > 0;
+}
+
+/**
+ * Get the current sweat amount
+ * @returns The amount of sweat
+ */
+export function sweat(): number {
+  return get("sweat");
+}
+
+/**
+ * Refresh the current sweat amount by visiting the item description
+ */
+export function refresh(): void {
+  visitUrl(`desc_item.php?whichitem=${item.descid}`);
```

Comment:
```suggestion
  examine(item);
```
importing examine from lib